app/full_text_search.py

```diff
@@ -149,22 +186,19 @@ def search(repo, query_txt, max_results):
         appengine_search.Query(
             query_string=and_query, options=options))
 
-    index_results = []
-    regexp = make_or_regexp(query_txt)
-    for document in person_location_index_results:
-        names = ''
-        romanized_jp_names = ''
-        for field in document.fields:
-            if field.name == 'names_romanized_by_romanize_word_by_unidecode':
-                names = field.value
-            if field.name == 'record_id':
-                id = field.value
-            if field.name == 'names_romanized_by_romanize_japanese_name_by_name_dict':
-                romanized_jp_names = field.value
-
-        if regexp.search(names) or regexp.search(romanized_jp_names):
-            index_results.append(id)
+    # For ranking same kanji higher than different kanji with the same reading.
```

Comment:
Now this is not limited to kanji, so you should instead say e.g., "To rank exact matches higher than non-exact matches with the same romanization".